test/test_iam_assume_token_manager.py

```diff
@@ -0,0 +1,192 @@
+def test_request_token_with_profile_name_and_account_id():
+    responses.add_callback(responses.POST, url=IAM_URL, callback=request_callback)
+
+    token_manager = IAMAssumeTokenManager("apikey", iam_profile_name=MY_PROFILE_NAME, iam_account_id=MY_ACCOUNT_ID)
+
+    # Make sure we don't have an access token yet.
+    assert token_manager.request_payload.get('access_token') is None
+
+    token_manager.request_token()
+
+    # Now the access token should be set along with the profile ID.
+    assert token_manager.request_payload.get('access_token') == ACCESS_TOKEN
+    assert token_manager.request_payload.get('profile_id') is None
+    assert token_manager.request_payload.get('profile_crn') is None
+    assert token_manager.request_payload.get('profile_name') == MY_PROFILE_NAME
+    assert token_manager.request_payload.get('account') == MY_ACCOUNT_ID
+
+
+@responses.activate
+def test_request_token_uses_the_correct_grant_types():
+    iam_url = "https://iam.cloud.ibm.com/identity/token"
+    responses.add(responses.POST, url=iam_url, body=BASE_RESPONSE_JSON, status=200)
+
+    token_manager = IAMAssumeTokenManager("apikey", iam_profile_id='my_profile_id')
+    token_manager.request_token()
+
+    assert token_manager.request_payload.get('grant_type') == 'urn:ibm:params:oauth:grant-type:assume'
+    assert token_manager.iam_delegate.request_payload.get('grant_type') == 'urn:ibm:params:oauth:grant-type:apikey'
+
+
+@responses.activate
+def test_request_token_uses_the_correct_headers():
+    responses.add_callback(responses.POST, url=IAM_URL, callback=request_callback)
+
+    token_manager = IAMAssumeTokenManager("apikey", iam_profile_id='my_profile_id')
+    token_manager.request_token()
+
+    assert len(responses.calls) == 2
+    assert responses.calls[0].request.headers.get('User-Agent').startswith('ibm-python-sdk-core/iam-authenticator')
+    assert (
+        responses.calls[1].request.headers.get('User-Agent').startswith('ibm-python-sdk-core/iam-assume-authenticator')
+    )
+
+
+@responses.activate
+def test_get_token():
+    responses.add_callback(responses.POST, url=IAM_URL, callback=request_callback)
+
+    token_manager = IAMAssumeTokenManager("apikey", iam_profile_id=MY_PROFILE_ID)
+
+    # Make sure we don't have an access token yet.
+    assert token_manager.request_payload.get('access_token') is None
+
+    access_token = token_manager.get_token()
+    # Now the access token should be set along with the profile ID.
+    assert token_manager.request_payload.get('access_token') == ACCESS_TOKEN
+    assert token_manager.request_payload.get('profile_id') == MY_PROFILE_ID
+    assert token_manager.request_payload.get('profile_crn') is None
+    assert token_manager.request_payload.get('profile_name') is None
+    assert token_manager.request_payload.get('account_id') is None
+
+    # The final result should be the other access token, which belong to the "assume" request.
+    assert access_token == OTHER_ACCESS_TOKEN
```

Comment:
Will you add an assertion that the refresh token is not set?

Reply:
Good catch. It turned out I only made this change in the authenticator and not in the token manager, so all these properties were accessible: `refresh_token`, `client_id`, `client_secret`

Reply:
So is it also the case that the client ID and secret will, if set, be used in the IAM delegate request but _not_ in the assume request?